src/libraries/System.Private.Xml/tests/TrimmingTests/System.Private.Xml.TrimmingTests.proj

```diff
@@ -2,12 +2,22 @@
   <Import Project="$([MSBuild]::GetPathOfFileAbove(Directory.Build.props))" />
 
   <ItemGroup>
-    <TestConsoleAppSourceFiles Include="XmlSchema.Write.cs" />
-    <TestConsoleAppSourceFiles Include="XmlSerializer.Deserialize.cs" />
+    <TestConsoleAppSourceFiles Include="XmlSchema.Write.cs">
+      <!-- What the test is testing is RUC and there was a half-hearted attempt to make it safe, but it isn't: https://github.com/dotnet/runtime/blob/5f067ce8b50087e032d13a1b97ae5ec39fc54739/src/libraries/System.Private.Xml/src/System/Xml/Schema/XmlSchema.cs#L134-L137 -->
+      <NativeAotIncompatible>true</NativeAotIncompatible>
+    </TestConsoleAppSourceFiles>
+    <TestConsoleAppSourceFiles Include="XmlSerializer.Deserialize.cs">
+      <NativeAotIncompatible>true</NativeAotIncompatible>
```

Comment:
Should we list a reason for `XmlSerializer.Deserialize.cs` and `XmlSerializer.Serialize.cs`?

Reply:
I struggled with what to put there and still struggle. I don't understand the reason we have this test in the first place. If we add a new optimization to IL Link that will make this test break same as it's broken with native AOT, would we consider that "illegal" optimization? What is the purpose of trimming tests for something that is marked as trim unsafe?

(My idea always was that we write trimming test if something is annotated as trim safe, but there's a warning suppression in place.)

Reply:
>  I don't understand the reason we have this test in the first place.

It is for Xamarin / Blazor WASM style trimming. See https://github.com/dotnet/runtime/issues/41389. Even though the API is marked as "unsafe" - we wanted to ensure it works if the developer does the work to preserve/mark their own types. (Sort of like how DiagnosticSource and EventSource work.) It is ensuring that we have our internals annotated correctly.

> I struggled with what to put there and still struggle.

What's the current problem we run into on native AOT? If we annotated in the test more, would that help?

Reply:
> What's the current problem we run into on native AOT? If we annotated in the test more, would that help?

There are two problems. I'm fixing one now in this PR, but the other problem requires #102482 (I put it into a separate PR so expect failures here for now).

Reply:
@eerhardt do you have a preference on how to proceed here? XmlSerializer doesn't seem to be actively maintained by anyone so nobody is looking at that PR and I would like to get this PR in so we can finally close the tracking issue.